streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala

```diff
@@ -122,17 +122,18 @@ object IngressController extends Logger {
   def ingressUrlAddress(nameSpace: String, clusterId: String, clusterClient: ClusterClient[_]): String = {
     if (determineIfIngressExists(nameSpace, clusterId)) {
       val client = new DefaultKubernetesClient
-      val ingress = client.network.ingress.inNamespace(nameSpace).withName(clusterId).get
-      val publicEndpoints = ingress.getMetadata.getAnnotations.get("field.cattle.io/publicEndpoints")
-      IngressMeta.as(publicEndpoints) match {
-        case Some(metas) =>
-          val ingressMeta = metas.head
-          val hostname = ingressMeta.hostname
-          val path = ingressMeta.path
-          logger.info(s"Retrieve flink cluster $clusterId successfully, JobManager Web Interface: https://$hostname$path")
-          s"https://$hostname$path"
-        case None => throw new RuntimeException("[StreamPark] get ingressUrlAddress error.")
-      }
+      // for kubernetes 1.22+
+      val fromV1 = Option(client.network.v1.ingresses.inNamespace(nameSpace).withName(clusterId).get)
+        .map(ingress => ingress.getSpec.getRules.get(0))
```

Comment:
We can `import scala.collection.JavaConversions._` and replace `.get(0)` to `.head`